app/Models/Game.php

```diff
@@ -120,6 +124,46 @@ protected static function newFactory(): GameFactory
         return GameFactory::new();
     }
 
+    public static function boot()
+    {
+        parent::boot();
+
+        static::pivotAttached(function ($model, $relationName, $pivotIds, $pivotIdsAttributes) {
+            if ($relationName === 'achievementSets') {
+                /** @var User $user */
+                $user = Auth::user();
+
+                activity()->causedBy($user)->performedOn($model)
+                    ->withProperty('old', [$relationName => null])
+                    ->withProperty('attributes', [$relationName => (new Collection($pivotIds))
+                        ->map(fn ($pivotId) => [
+                            'id' => $pivotId,
+                            'attributes' => $pivotIdsAttributes[$pivotId],
+                        ]),
+                    ])
+                    ->event('pivotAttached')
+                    ->log('pivotAttached');
+            }
+        });
```

Comment:
Are the audit logs supposed to be raw?
![image](https://github.com/user-attachments/assets/70691db0-24be-42e5-9f32-501b8b20b6bd)

Reply:
Yes. This caught me by surprise too. At first, these elements crashed the page altogether because they don't have a `->name` field. After I fixed that, they fall into the next branching case, which is to just dump the JSON:
```php
{{-- audit-log.blade.php --}}

@elseif (is_array($newValue))
    <pre class="text-xs text-gray-500">{{ json_encode($newValue, JSON_PRETTY_PRINT | JSON_UNESCAPED_SLASHES) }}</pre>
```
I started working to enhance this to look a bit better, but I quickly realized that the complexity of doing so would be kinda high. May be worth revisiting later.